h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -166,15 +172,36 @@ public ParseSetup(ParseSetupV3 ps) {
    */
   public ParseSetup(ParserInfo parseType, byte sep, boolean singleQuotes, int checkHeader,
                     int ncols, String[] columnNames, byte[] ctypes,
-                    String[][] domains, String[][] naStrings, String[][] data){
+                    String[][] domains, String[][] naStrings, String[][] data, byte[] nonDataLineMarkers) {
     this(parseType, sep, singleQuotes, checkHeader, ncols, columnNames, ctypes,
-        domains, naStrings, data, new ParseWriter.ParseErr[0], FileVec.DFLT_CHUNK_SIZE);
+        domains, naStrings, data, new ParseWriter.ParseErr[0], FileVec.DFLT_CHUNK_SIZE, nonDataLineMarkers);
   }
+
+  /**
+   * Create a ParseSetup with all parameters except chunk size.
+   *
+   * Typically used by file type parsers for returning final valid results
+   * _chunk_size will be set later using results from all files.
+   */
+  public ParseSetup(ParserInfo parseType, byte sep, boolean singleQuotes, int checkHeader,
```

Comment:
this is becoming a constructor hell :)

Reply:
builder?

Reply:
Thanks for pointing it out. Turns out the constructor injection can be "eased" by deleting some of the older ones. Those were not marked as unused code, but the ones that remained after commit https://github.com/h2oai/h2o-3/pull/3380/commits/0e610d07febddbc27082c31202d9bbbc4ddcbe7b are completely OK.